### DISCUSSION TITLE
flowmap arrow

### DISCUSSION THREAD

**Discussion by sgavathe** [FlowmapBlue/FlowmapBlue#39] [Q&A] [answered]
Hello,

Anyone know how to draw the flowmap directional arrow on both side of the line instead of half arrow ? 

![image (36)](https://user-images.githubusercontent.com/6047188/130071152-9a6fc00f-9b62-4384-a91a-d464f9f3506a.png)

![image (37)](https://user-images.githubusercontent.com/6047188/130071171-e690641f-1c4c-4478-9607-76c2e4a3c835.png)

**Comment by ilyabo** [comment, marked as answer]
Hi Shailesh,

You just need to include the reverse direction flow into your data. So if you only have some count for the pair of origin A and dest B, you need to add another row into your flows dataset with the origin B, dest C and some non-zero count. 

Does this answer your question?

**Reply by sgavathe** [reply]
Thank you Ilya, appreciated it . 

Will that draw a full arrow like below ?
![image](https://user-images.githubusercontent.com/6047188/130080500-af99e8d7-d709-4574-8a00-11cdd3bf5f71.png)

Or do I need to adjust the positions ?

  ```
if (drawOutline) {
                // source_target_mix, perpendicular_offset_in_thickness_units, direction_of_travel_offset_in_thickness_units
                // prettier-ignore
                positions = positions.concat([
                        // Outline
                        0, 0, 0,
                        0, 1, 0,
                        1, 0, 0,

                        0, 1, 0,
                        1, 0, -3,
                        1, 1, -3,

                        1, 0, 0,
                        1, 2, -3,
                        //1, 0, -3,
                        1, -2, -3
                    ],
                );
```

**Reply by ilyabo** [reply]
Ah, now I see what you mean. Yes, if you need to change the look of the arrow, you need to adjust the positions, also you'll need to add two more vertices, I think. Every row in these array corresponds to one vertex of an arrow, and the values specify where it should be positioned.

**Reply by sgavathe** [reply]
Thank you Ilya.

I was struggling to see how the positions are laid out. Does each of these mean vertices ? Sorry, not too familiar with the positions and normals. 

`  0, 0, 0,
   0, 1, 0,
   1, 0, 0,`

**Reply by ilyabo** [reply]
This is something very custom. Each three numbers are for one vertex.

**Reply by ilyabo** [reply]
The three values for one vertex have these meanings respectively to displace the vertex:

    source_target_mix, perpendicular_offset_in_thickness_units, direction_of_travel_offset_in_thickness_units

**Reply by ilyabo** [reply]
`source_target_mix` means the displacement from the source towards the target

**Reply by sgavathe** [reply]
ahh got you  Ilya.

Makes sense on how positions are done for vertex and how numbers representing for a shape. Thanks a lot.
